tools/deps_info.py

```diff
@@ -212,6 +210,8 @@ def get_deps_info():
     _deps_info['__cxa_find_matching_catch_7'] = ['__cxa_can_catch']
     _deps_info['__cxa_find_matching_catch_8'] = ['__cxa_can_catch']
     _deps_info['__cxa_find_matching_catch_9'] = ['__cxa_can_catch']
+  if settings.FILESYSTEM and settings.SYSCALLS_REQUIRE_FILESYSTEM:
+    _deps_info['mmap'] = ['emscripten_builtin_memalign']
```

Comment:
this can probably be unconditional - if you use mmap, you surely use more filesystem code. `!SYSCALLS_REQUIRE_FILESYSTEM` mode basically means "pure computation, no files, no overhead from files".

Reply:
That doesn't work given the way we test the deps_info.

We create a tiny program that just does `int main() { return &mmap; }`.. in that program mmap is used but the  `emscripten_builtin_memalign` dependency does not show up.. so the entry in deps_info would be reported (correctly as an invalid dependency).

Reply:
> this can probably be unconditional - if you use mmap, you surely use more filesystem code. `!SYSCALLS_REQUIRE_FILESYSTEM` mode basically means "pure computation, no files, no overhead from files".

Actually lots of program use mmap just to allocate memory (`MAP_ANON`) without any need for a filesystem.

Reply:
I see, sgtm